x-pack/plugins/triggers_actions_ui/public/application/sections/rules_list/components/rule_status_dropdown.tsx

```diff
@@ -0,0 +1,471 @@
+  onClosePopover: () => void;
+  isEnabled: boolean;
+  isSnoozed: boolean;
+  snoozeEndTime?: Date | null;
+}
+
+const RuleStatusMenu: React.FunctionComponent<RuleStatusMenuProps> = ({
+  onChangeEnabledStatus,
+  onChangeSnooze,
+  onClosePopover,
+  isEnabled,
+  isSnoozed,
+  snoozeEndTime,
+}) => {
+  const enableRule = useCallback(() => {
+    if (isSnoozed) {
+      // Unsnooze if the rule is snoozed and the user clicks Enabled
+      onChangeSnooze(0, 'm');
+    } else {
+      onChangeEnabledStatus(true);
+    }
+    onClosePopover();
+  }, [onChangeEnabledStatus, onClosePopover, onChangeSnooze, isSnoozed]);
+  const disableRule = useCallback(() => {
+    onChangeEnabledStatus(false);
+    onClosePopover();
+  }, [onChangeEnabledStatus, onClosePopover]);
+
+  const onApplySnooze = useCallback(
+    (value: number, unit?: SnoozeUnit) => {
+      onChangeSnooze(value, unit);
+      onClosePopover();
+    },
+    [onClosePopover, onChangeSnooze]
+  );
+
+  let snoozeButtonTitle = <EuiText size="s">{SNOOZE}</EuiText>;
+  if (isSnoozed && snoozeEndTime) {
+    snoozeButtonTitle = (
+      <>
+        <EuiText size="s">{SNOOZE}</EuiText>{' '}
+        <EuiText size="xs" color="subdued">
+          {moment(snoozeEndTime).format(SNOOZE_END_TIME_FORMAT)}
+        </EuiText>
+      </>
+    );
+  }
+
+  const panels = [
+    {
+      id: 0,
+      width: 360,
+      items: [
+        {
+          name: ENABLED,
+          icon: isEnabled && !isSnoozed ? 'check' : 'empty',
+          onClick: enableRule,
+        },
+        {
+          name: DISABLED,
+          icon: !isEnabled ? 'check' : 'empty',
+          onClick: disableRule,
+        },
+        {
+          name: snoozeButtonTitle,
+          icon: isEnabled && isSnoozed ? 'check' : 'empty',
+          panel: 1,
+          disabled: !isEnabled,
+        },
+      ],
+    },
+    {
+      id: 1,
+      width: 360,
+      title: SNOOZE,
+      content: (
+        <SnoozePanel
+          applySnooze={onApplySnooze}
+          interval={futureTimeToInterval(snoozeEndTime)}
+          showCancel={isSnoozed}
+        />
+      ),
+    },
+  ];
+
+  return <EuiContextMenu initialPanelId={0} panels={panels} />;
+};
+
+interface SnoozePanelProps {
+  interval?: string;
+  applySnooze: (value: number | -1, unit?: SnoozeUnit) => void;
+  showCancel: boolean;
+}
+
+const SnoozePanel: React.FunctionComponent<SnoozePanelProps> = ({
+  interval = '3d',
+  applySnooze,
+  showCancel,
+}) => {
+  const [intervalValue, setIntervalValue] = useState(parseInterval(interval).value);
+  const [intervalUnit, setIntervalUnit] = useState(parseInterval(interval).unit);
+
+  const onChangeValue = useCallback(
+    ({ target }) => setIntervalValue(target.value),
+    [setIntervalValue]
+  );
+  const onChangeUnit = useCallback(
+    ({ target }) => setIntervalUnit(target.value),
+    [setIntervalUnit]
+  );
+
+  const onApply1h = useCallback(() => applySnooze(1, 'h'), [applySnooze]);
```

Comment:
It feels weird to hard code it like this - is it possible to define the commonly used time frames in some constant at the top and then build out the UI from that list?

Reply:
From an i18n perspective, that could work if we were okay with Moment.js converting `1h` to `an hour` instead of `1 hour` but I don't think we are? Unfortunately I'm not sure how to customize Moment's `humanize` function to use `1` instead of `an` (or equivalent, depending on locale; `fr` uses `une minute` for example).

If we're okay with changing the language then this would work but otherwise we're constrained by i18n.